### PR TITLE
Wip/hughsie/fwupdtool non root2

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -5929,14 +5929,10 @@ fu_engine_load_plugins (FuEngine *self, GError **error)
...
 				  self);
-
-		/* if loaded from fu_engine_load() open the plugin */
-		if (self->usb_ctx != NULL) {
-			if (!fu_plugin_open (plugin, filename, &error_local)) {
-				g_warning ("cannot load: %s", error_local->message);
-				fu_engine_add_plugin (self, plugin);
-				continue;
-			}
+		if (!fu_plugin_open (plugin, filename, &error_local)) {
```
Comment (superm1) [inline src/fu-engine.c:5932]: the check for `NULL` was specifically to support the `get-plugins` command for fwupdtool.

Comment (superm1) [reply src/fu-engine.c:5932]: Here's the commit it came from: https://github.com/fwupd/fwupd/commit/8c71a3f2e98262d672f90fe0bcc0841e550083cc

File: src/fu-engine.c [hunk 2]
```diff
@@ -6289,6 +6289,48 @@ fu_engine_ensure_client_certificate (FuEngine *self)
...
 
+static void
+fu_engine_add_firmware_gtypes (FuEngine *self)
+{
+	fu_engine_add_firmware_gtype (self, "raw", FU_TYPE_FIRMWARE);
+	fu_engine_add_firmware_gtype (self, "dfu", FU_TYPE_DFU_FIRMWARE);
+	fu_engine_add_firmware_gtype (self, "fmap", FU_TYPE_FMAP_FIRMWARE);
+	fu_engine_add_firmware_gtype (self, "ihex", FU_TYPE_IHEX_FIRMWARE);
+	fu_engine_add_firmware_gtype (self, "srec", FU_TYPE_SREC_FIRMWARE);
+}
```
Comment (superm1) [inline src/fu-engine.c:6300]: a static function only ever called only once for 5 different void functions?  Seems like it should just be in the caller function.

Comment (superm1) [reply src/fu-engine.c:6300]: oh never mind, It's two code paths now, I missed that. sorry for the noise.

File: src/fu-tool.c [hunk 3]
```diff
@@ -1748,7 +1748,7 @@ fu_util_get_firmware_types (FuUtilPrivate *priv, gchar **values, GError **error)
 	g_autoptr(GPtrArray) firmware_types = NULL;
 
 	/* load engine */
-	if (!fu_engine_load (priv->engine, FU_ENGINE_LOAD_FLAG_NO_ENUMERATE, error))
+	if (!fu_engine_load_firmware (priv->engine, error))
```
Comment (superm1) [inline src/fu-tool.c:1751]: why not have a flag that `FU_ENGINE_LOAD_FIRMWARE_ONLY` you detect and keep the function the same?

Comment (hughsie) [reply src/fu-tool.c:1751]: Alternate proposal: https://github.com/fwupd/fwupd/pull/2670

**Pull request closed without merging**